src/js/core/textual_game_state.tsx

```diff
@@ -1,40 +1,76 @@
 import { HUDModalDialogs } from "../game/hud/parts/modal_dialogs";
 import { GameState } from "./game_state";
-import { T } from "../translations";
 
 /**
  * Baseclass for all game states which are structured similary: A header with back button + some
  * scrollable content.
  */
-export class TextualGameState extends GameState {
-    ///// INTERFACE ////
+export abstract class TextualGameState extends GameState {
+    private backToStateId: string | null = null;
+    private backToStatePayload: {} | null = null;
+
+    protected headerElement: HTMLElement;
+    protected containerElement: HTMLElement;
+    protected dialogs: HUDModalDialogs;
 
     /**
      * Should return the states inner html. If not overriden, will create a scrollable container
      * with the content of getMainContentHTML()
-     * @returns {string}
+     * @deprecated
      */
-    getInnerHTML() {
-        return `
-            <div class="content mainContent">
-                ${this.getMainContentHTML()}
-            </div>
-        `;
+    getInnerHTML(): string {
+        return "";
     }
 
     /**
      * Should return the states HTML content.
+     * @deprecated
      */
-    getMainContentHTML() {
+    getMainContentHTML(): string {
         return "";
     }
 
+    protected override getContentLayout(): Node {
+        let content = this.getInitialContent();
+
+        if (content === null) {
+            // Fall back either to getMainContentHTML or getInnerHTML (if not "")
+            let html = this.getInnerHTML();
```

Comment:
feels a bit weird that this fallback to `getInnerHTML()` uses a strict comparison with null, unlike in game state

Reply:
There are a few minor differences between the default behavior of GameState and TextualGameState. Honestly, I don't really remember why this works like it does (there are already differences between layout generated from HTML strings and from getInitialContent because I was so confused!), but what I know is that GameState needs to handle all kinds of return values so the rest of the checks can replace the simpler null check. I just committed a change that also makes getContentLayout never return null in the default GameState implementation as well.